File: work-dispatcher/src/main/java/works/service/WorkUnitGateway.java
package works.service;

import org.springframework.integration.annotation.Gateway;
import org.springframework.integration.annotation.MessagingGateway;
import works.service.domain.WorkUnit;

@MessagingGateway
public interface WorkUnitGateway {
	@Gateway(requestChannel = WorkUnitsSource.CHANNEL_NAME)
	void generate(WorkUnit workUnit);

}

File: work-dispatcher/src/main/kotlin/works/service/WorkUnitGateway.kt
package works.service

import org.springframework.integration.annotation.Gateway
import org.springframework.integration.annotation.MessagingGateway
import works.service.domain.WorkUnit

@MessagingGateway
interface WorkUnitGateway {
    @Gateway(requestChannel = WorkUnitsSource.CHANNEL_NAME)
    fun generate(workUnit: WorkUnit)

}
